tests/unit/test_charm.py

```diff
@@ -1976,6 +1976,47 @@ def test_ingresses_for_multiple_relations_blocked(
         self.assertEqual("active", self.harness.charm.unit.status.name)
         self.assertIn(expected_status_message, self.harness.charm.unit.status.message)
 
+    @patch("charm.NginxIngressCharm._report_ingress_ips")
+    @patch("charm.NginxIngressCharm._report_service_ips")
+    @patch("charm.NginxIngressCharm._define_service")
+    @patch("charm.NginxIngressCharm._networking_v1_api")
+    def test_ingresses_for_invalid_hostname(
```

Comment:
There is an interesting case here @gregory-schiano 

I'm guessing that the hostname regex came from an RFC so it should be well tested. However, we do write the regex into our code and essentially take on ownership that it is accurate.

Since it is essentially vendoring in the regex, should we add a few more tests, potentially using https://docs.pytest.org/en/6.2.x/parametrize.html?

Reply:
Well indeed we take ownership of this regex as it's in our code, and there's a risk of modification of the regex in a future PR
So we need to indeed test with more cases.
Ideally this regex should come from a library which purpose is domains.

On the other side we probably shouldn't bo too  deep in the cases we cover as if there's one case we don't cover, the consequences aren't too big

Reply:
The only library I found that validates FQDNs is [fqdn](https://github.com/ypcrts/fqdn) but its validation is not case sensitive, so it won't be useful for us. I'll proceed to add more tests.